apps/src/templates/studioHomepages/JoinSection.jsx

```diff
@@ -140,7 +177,14 @@ export default class JoinSection extends React.Component {
 
   render() {
     const {enrolledInASection} = this.props;
+    const {isLoaded, sectionAttempts} = this.state;
 
+    //captcha verification required when more than 3 section attempts in 24 hours
+    const requireCaptcha = sectionAttempts >= 3;
```

Comment:
Should requireCaptcha be set by the server instead? i.e. I think with the way the code is right now, I could refresh the page every 2 attempts and the dialog would never be displayed.

Reply:
`sectionAttempts` is pulled in from the server on page load. When you send a `POST` request to the backend to join a section, that is all done asynchronously, so in order to keep the backend data in sync with the counter on the frontend (`this.state.sectionAttempts`), we increment `this.state.sectionAttempts` each time the user attempts to join a section. Refreshing the page would reset the state to zero, but it would then be overwritten by the value fetched from the database. Does that make sense?

Reply:
Ahh I see. Thanks for the clarification! ✔️

Reply:
I think it'd probably be to note this in a comment where you increment this.state.sectionAttempts.